File: spec/javascript/test_debug_0.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('Debug0', 'src/fixed_struct.bin', function(r) {
  r._read();

  assert.strictEqual(r.one, 80);
  assert.deepStrictEqual(r.arrayOfInts, [65, 67, 75]);
  assert.strictEqual(r._unnamed2, 45);

  assert.deepStrictEqual(r._debug, {
      "one": {
          "start": 0,
          "ioOffset": 0,
          "end": 1
      },
      "arrayOfInts": {
          "start": 1,
          "ioOffset": 0,
          "arr": [
              {
                  "start": 1,
                  "ioOffset": 0,
                  "end": 2
              },
              {
                  "start": 2,
                  "ioOffset": 0,
                  "end": 3
              },
              {
                  "start": 3,
                  "ioOffset": 0,
                  "end": 4
              }
          ],
          "end": 4
      },
      "_unnamed2": {
          "start": 4,
          "ioOffset": 0,
          "end": 5
      }
  });
});

File: spec/lua/test_debug_0.lua
local luaunit = require("luaunit")

require("debug_0")

TestDebug0 = {}

function TestDebug0:test_debug_0()
    local r = Debug0:from_file("src/fixed_struct.bin")

    -- --debug implies --no-auto-read
    r:_read()

    luaunit.assertEquals(r.one, 80)
    luaunit.assertEquals(#r.array_of_ints, 3)
    luaunit.assertEquals(r.array_of_ints[0 + 1], 65)
    luaunit.assertEquals(r.array_of_ints[1 + 1], 67)
    luaunit.assertEquals(r.array_of_ints[2 + 1], 75)
    luaunit.assertEquals(r._unnamed2, 45)

    -- FIXME: also test --read-pos once it is implemented
end
